dolphin/api/v1/alert.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from oslo_log import log
+from webob import exc
+
+from dolphin import db, cryptor
+from dolphin import exception
+from dolphin import utils
+from dolphin.api import validation
+from dolphin.api.common import wsgi
+from dolphin.api.schemas import alert as schema_alert
+from dolphin.api.views import alert as alert_view
+from dolphin.drivers import api as driver_api
+from dolphin.i18n import _
+
+LOG = log.getLogger(__name__)
+
+
+SNMPv3_keys = ('username', 'password', 'auth_enable', 'auth_protocol', 'encryption_enable',
+               'encryption_protocol', 'encryption_password', 'engine_id')
+
+
+class AlertController(wsgi.Controller):
+    def __init__(self):
+        super().__init__()
+        self.driver_api = driver_api.API()
+
+    @staticmethod
+    def exist(ctx, storage_id):
+        try:
+            db.alert_source_get(ctx, storage_id)
+        except exception.AlertSourceNotFound:
+            return False
+
+        return True
+
+    @staticmethod
+    def decrypt_password(alert_source):
+        password = alert_source.get('password', None)
+        encryption_password = alert_source.get('encryption_password', None)
```

Comment:
not able to understand password and encryption_password diff

Reply:
> not able to understand password and encryption_password diff

Plz see Table 1 from https://www.cisco.com/c/en/us/td/docs/ios-xml/ios/snmp/configuration/xe-3se/3850/snmp-xe-3se-3850-book/nm-snmp-snmpv3.html.